Sources/MessagingInApp/Gist/Gist.swift

```diff
@@ -32,7 +32,11 @@ public class Gist: GistDelegate {
     func reset() {
         clearUserToken()
         messageQueueManager = MessageQueueManager()
-        messageManagers = []
+
+        for messageManager in messageManagers {
+            removeMessageManager(instanceId: messageManager.currentMessage.instanceId)
+        }
```

Comment:
After mocking animations, I was getting crashes in the test suite when `messageManagers = []` was running. The error message was a memory access error because `messageManagers` was being accessed while clearing it. 

After debugging the error, I decided to only modify the test suite code instead of making a refactor to the SDK code and possibly break a feature in the SDK.